### PR TITLE
Refactor MAP container to easily scale to new components

### PR DESCRIPTION
**Pull request by dr3**
**Overall change:** Based on https://github.com/bbc/simorgh/pull/3719, make the MAP container easily scaleable 

**Code changes:**

- Split blocks out into their own folders
- Split block looping into its own container

Testing notes: 
Refactor, shouldn't cause any changes to page behaviour or layout. Basic regression on MAP and Live Radio routes

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [x] This PR requires manual testing

### CODE REVIEW

File: src/app/containers/MediaPageBlocks/Blocks/Unknown/index.jsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import { string } from 'prop-types';
+
+/* eslint-disable react/no-danger */
+const UnknownBlockContainer = ({ blockType }) => (
+  <div
+    dangerouslySetInnerHTML={{ __html: `<!-- unknown block ${blockType} -->` }}
```
Comment (ryanmccombe) [inline src/app/containers/MediaPageBlocks/Blocks/Unknown/index.jsx]: benefit of adding the type to this comment is probably not worth the xss risk

**Review by ryanmccombe — APPROVED** [review]
Not too sure we need to have all these id props?

Looks good though - thanks for doing this!

File: src/app/containers/MediaPageBlocks/Blocks/Unknown/index.jsx (hunk 1 above)
Comment (dr3) [reply src/app/containers/MediaPageBlocks/Blocks/Unknown/index.jsx]: Hmm, I get your point, but the data can only be from the page data, is that much of a risk?

Comment (ryanmccombe) [reply src/app/containers/MediaPageBlocks/Blocks/Unknown/index.jsx]: doesn't the page data come from external systems though?  cps / ares

the risk is certainly small, but the benefit is even smaller :)

File: src/app/containers/MediaPageBlocks/Blocks/LiveRadio/index.test.jsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+describe('MediaPageBlocks LiveRadio', () => {
+  shouldMatchSnapshot(
+    'should render correctly for canonical',
+    <RequestContext.Provider value={{ platform: 'canonical' }}>
+      <LiveRadio uuid="uuid" idAttr="idAttr" externalId="externalId" id="id" />
+    </RequestContext.Provider>,
+  );
+
+  // TODO: remove the need for this suppressPropWarnings
```
Comment (pjlee11) [inline src/app/containers/MediaPageBlocks/Blocks/LiveRadio/index.test.jsx:15]: 👀

**Review by pjlee11 — APPROVED** [review]
LGTM, I think some of these blocks should be refactored to better reuse some of the optimo containers but that's follow up work.

File: src/app/containers/MediaPageBlocks/Blocks/Heading/index.jsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import React, { useContext } from 'react';
+import { string } from 'prop-types';
+import { Headline } from '@bbc/psammead-headings';
+import { ServiceContext } from '#contexts/ServiceContext';
+
+const HeadingContainer = ({ idAttr, text }) => {
+  const { script, service } = useContext(ServiceContext);
+
+  if (!text) return null;
```
Comment (jroebu14) [inline src/app/containers/MediaPageBlocks/Blocks/Heading/index.jsx:9]: It'd be good to have a discussion about whether these checks should happen in the component or in the parent component. I think it should be done in the parent as it's kind of container-y logic e.g. `headline && <Headline>` and makes this component more like a dumb and logic-less template that doesn't need the extra null test cases where it cheats the prop types validation. Would be keen to know what others think though.

File: src/app/containers/MediaPageBlocks/Blocks/Unknown/index.jsx (hunk 1 above)
Comment (jroebu14) [reply src/app/containers/MediaPageBlocks/Blocks/Unknown/index.jsx]: I agree with Ryan on this

**Review by jroebu14 — APPROVED** [review]
Looks good but would like to know what you think about https://github.com/bbc/simorgh/pull/3738#discussion_r326518872

File: src/app/containers/MediaPageBlocks/Blocks/Heading/index.jsx (hunk 3 above)
Comment (dr3) [reply src/app/containers/MediaPageBlocks/Blocks/Heading/index.jsx:9]: @jroebu14 IMO All blocks should deal with their own logic. We call all of them unbiased and then they themselves decide if they render or not. Keeps logic isolated

**Comment by LilyL0u** [comment]
These are how the pages are looking after this change:

Image 20-09-2019 at 16 01

Image 20-09-2019 at 16 01

The risk here is that, because we cannot see the player component locally at the moment, we can't test if that has been affected by the refactoring. Would it be possible to get this onto test before marking this as passed testing? 
@dr3 @ryanmccombe @pjlee11 (would the player be affected by these changes anyway?)
Or do we accept this risk? @graemelindsayBBC 

The MAP page is minimal but looks the same as before.

**Comment by ryanmccombe** [comment]
This shouldn't touch the player, and seems safe to merge because our pages aren't live yet anyway

Having to merge and make code live to see changes sucks, but Mozart should have the work done to get the players to work locally before our pages are live.  So this process should only be temporary.

**Comment by LilyL0u** [comment]
I will mark this as passed testing, as the pages look the same as before the refactor, and the refactor should not touch the player.

**Review by chris-hinds — APPROVED** [review]

**Pull request merged**